lib/core/notification/notification_service.dart

```diff
@@ -110,7 +110,26 @@ class NotificationService implements INotificationService {
       );
 
   @override
-  Future<void> init() async {
+  Future<void> init({bool isApnsTokenRetryAttempt = false}) async {
+    if (Platform.isIOS) {
+      final apnsToken = await FirebaseMessaging.instance.getAPNSToken();
+      if (apnsToken == null && isApnsTokenRetryAttempt != true) {
+        // sometimes it takes a while to get the apns token on startup
+        await Future.delayed(
+          const Duration(seconds: 1),
+          () async => init(
+            isApnsTokenRetryAttempt: true,
+          ),
+        );
+      } else {
```

Comment:
Instead of having else statements and multiple places with doinit you could call _doInit() once at the end of this method and do a return after the Future.delayed (the recursive loop)